cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -0,0 +1,462 @@
+
+        ebs_metrics = [
+            GeneralMetric("Read/Write Ops", ["VolumeReadOps", "VolumeWriteOps"]),
+            GeneralMetric("Read/Write Bytes", ["VolumeReadBytes", "VolumeWriteBytes"]),
+            GeneralMetric("Total Read/Write Time", ["VolumeTotalReadTime", "VolumeTotalWriteTime"]),
+            GeneralMetric("Queue Length", ["VolumeQueueLength"]),
+            GeneralMetric("Idle Time", ["VolumeIdleTime"]),
+        ]
+
+        conditional_ebs_metrics = [
+            GeneralMetric("Consumed Read/Write Ops", "VolumeConsumedReadWriteOps", ["io1", "io2", "gp3"]),
+            GeneralMetric("Throughput Percentage", "VolumeThroughputPercentage", ["io1", "io2", "gp3"]),
+            GeneralMetric("Burst Balance", "BurstBalance", ["gp2", "st1", "sc1"]),
+        ]
+
+        # Add EBS and RAID metrics graphs
+        for storage_type, title in [
+            (SharedStorageType.EBS, "## EBS Metrics"),
+            (SharedStorageType.RAID, "## RAID Metrics"),
+        ]:
+            if len(self.storage_resources[storage_type]) > 0:
+                self._add_volume_metrics_graphs(title, storage_type, ebs_metrics, conditional_ebs_metrics)
+
+        # Add EFS metrics graphs
+        if len(self.storage_resources[SharedStorageType.EFS]) > 0:
+            self._add_efs_metrics_graphs()
+
+        # Add FSx metrics graphs
+        if len(self.storage_resources[SharedStorageType.FSX]) > 0:
+            self._add_fsx_metrics_graphs()
+
+        # Head Node logs, if CW Logs are enabled
+        if self.config.is_cw_logging_enabled:
+            self._add_cw_log()
+
+    def _update_coord(self, d_x, d_y):
+        """Calculate coordinates for the new graph."""
+        self.coord.x_value = self.coord.x_value + d_x
+        if self.coord.x_value > 24:  # the longest width allowed is 24
+            self.coord = Coord(0, self.coord.y_value + d_y)  # updates both x and y values
+
+    def _update_coord_after_section(self, d_y):
+        """Calculate section header coordinates."""
+        self.coord.y_value = self.coord.y_value + d_y
+
+    def _reset_coord(self):
+        self.coord = Coord(0, 0)
+
+    def _is_logs_section_empty(self, section_widgets):
+        """Check if a section will be empty."""
+        self.empty_section = True
+        for log_params in section_widgets.widgets:
+            if log_params.conditions:
+                for cond_dict in log_params.conditions:
+                    if cond_dict.param in cond_dict.allowed_values:
+                        self.empty_section = False
+
+    def _add_text_widget(self, markdown):
+        """Add the textwidget to the cloudwatch dashboard and update coordinates."""
+        text_widget = cloudwatch.TextWidget(markdown="\n" + markdown + "\n", height=1, width=24)
+        text_widget.position(x=self.coord.x_value, y=self.coord.y_value)
+        self.cloudwatch_dashboard.add_widgets(text_widget)
+        self._update_coord_after_section(1)
```

Comment:
let's use kwargs, it helps to understand what is this 1 you're passing to the function.
`self._update_coord_after_section(d_y=1)`